File: 1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var i
    var dic = {}
    var ans = new Array(2)
    for(i = 0; i < nums.length; i++) {
        if(dic[target - nums[i]] != undefined){
            ans[0] = dic[target - nums[i]]
            ans[1] = i + 1
            break
        }
        dic[nums[i]] = i + 1
    }
    return ans
};

File: 1.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dic = {}
        n = len(nums)
        for i in xrange(0, n):
            temp = target - nums[i]
            if dic.has_key(temp):
                a = dic[temp]
                b = i + 1
                break
            dic[nums[i]] = i + 1
        
        return a,b
